system/modules/core/library/Contao/Model/Registry.php

```diff
@@ -83,16 +89,49 @@ public function count()
 	/**
 	 * Fetch a model by table name and primary key
 	 *
-	 * @param string  $strTable The table name
-	 * @param integer $intPk    The primary key
+	 * @param string  $strTable     The table name
+	 * @param mixed   $varKey       The key
+	 * @param string  $strAlias     An optional alias
 	 *
 	 * @return \Model|null The model or null
 	 */
-	public function fetch($strTable, $intPk)
+	public function fetch($strTable, $varKey, $strAlias = null)
 	{
-		if (isset($this->arrRegistry[$strTable][$intPk]))
+		// Default is searching by PK and is the most common case
```

Comment:
I know we are in legacy code here and not in Contao 4 but please please please let's not increase the complexity of the old classes if we can use methods for new code even if it is in legacy classes.